### PR TITLE
QueueModal:UI/UX

### PR DESCRIPTION
**Pull request by maelswarm**
The key changes have to do with the the reducer for action method ```mintTokenAction```.

```mintToken``` is no longer type ```Status```, but type ```Array<Status>```.

This allows us to use the ```meta.requestId``` as a unique Id within the array of Status objects/types.

### CODE REVIEW

**Comment by wyc** [comment]
Thanks, this is a great direction. It worked locally for me. A few suggestions:
- The modal seems to indicate `in_transit` before it has been signed and sent, such as loading the Temple password page. Can we update this to be more accurate, only after signing and sending to the RPC?
- Is it possible to redirect back to the main page after clicking "create"? Users will probably want to do something else while their tx is clearing.
- If the user refreshes, could we use localStorage to ensure we're persisting the state?

**Comment by maelswarm** [comment]
- The modal seems to indicate in_transit before it has been signed and sent, such as loading the Temple password page. Can we update this to be more accurate, only after signing and sending to the RPC?


Yes

Is it possible to redirect back to the main page after clicking "create"? Users will probably want to do something else while their tx is clearing.


Yes

- If the user refreshes, could we use localStorage to ensure we're persisting the state?

Sort of. This will require other stuff.


@wyc

File: src/reducer/slices/status.ts [hunk 1]
```diff
@@ -22,11 +21,12 @@ export interface Status {
     rejectValue: RejectValue;
     serialized: SerializedError;
   } | null;
+  contract?: String
```
Comment (ricklove) [inline src/reducer/slices/status.ts:24]: `String` as a type should be lowercase `string`

File: src/reducer/slices/status.ts [hunk 2]
```diff
@@ -52,22 +52,35 @@ const initialState: StatusState = {
...
+        } else {
+          state[payload.method] = (state[payload.method]).map(c => {
+            if(c.contract === payload.contract) {
+              c.status = payload.status;
+            }
+            return c;
+          }) as Array<Status>;
+        }
+      } else {
+        (state[payload.method] as any).status = payload.status;
```
Comment (ricklove) [inline src/reducer/slices/status.ts:79]: Why is there `as any` here?

File: src/reducer/slices/status.ts [hunk 3]
```diff
@@ -82,23 +95,58 @@ const slice = createSlice({
...
+          } else {
+            state[method] = (state[method] as Array<Status>).map(c => {
+              if(c.contract === a?.meta?.requestId) {
+                c.status = 'in_transit';
+              }
+              return c;
+            }) as Array<Status>;
+          }
+        } else {
+          (state[method] as any).status = 'in_transit';
```
Comment (ricklove) [inline src/reducer/slices/status.ts:111]: Also here, `as any` is not needed. The type seems to be known.

Comment (ricklove) [reply src/reducer/slices/status.ts:111]: (And same below in the other cases where as any was added)

File: src/components/CreateNonFungiblePage/index.tsx [hunk 4]
```diff
@@ -145,25 +144,12 @@ export default function CreateNonFungiblePage() {
...
-                dispatch(mintTokenAction());
-              }}
-              onCancel={() => {
-                onClose();
-                dispatch(clearError({ method: 'mintToken' }));
-                dispatch(setStatus({ method: 'mintToken', status: 'ready' }));
-              }}
-              status={status}
-            />
+            <div style={{position: 'absolute', bottom: 0, right: 0, width: 'auto', display: 'flex', flexFlow: 'column nowrap', justifyContent:'flex-end', alignItems: 'flex-end', borderTopRightRadius: '5px', boxShadow: '-2px -2px 20px #3333', padding: '5px 0 0 5px'}}>
```
Comment (ricklove) [inline src/components/CreateNonFungiblePage/index.tsx:147]: Bug: This doesn't navigate back to the collection after the creation of the token anymore.

**Review by ricklove — CHANGES_REQUESTED** [review]
Unintended behavior change? and type issues.

File: src/reducer/slices/status.ts (hunk 1 above)
Comment (maelswarm) [reply src/reducer/slices/status.ts:24]:
```
Type 'String | undefined' is not assignable to type 'string | undefined'.
  Type 'String' is not assignable to type 'string'.
    'string' is a primitive, but 'String' is a wrapper object. Prefer using 'string' when possible.  TS2322
```

I get that when I try to use ```string```

**Comment by maelswarm** [comment]
@wyc 
Should redirect to /create location now.

Other changes are going to need more of an overhaul.

File: src/components/CreateNonFungiblePage/index.tsx (hunk 4 above)
Comment (maelswarm) [reply src/components/CreateNonFungiblePage/index.tsx:147]: I'm not sure I follow, it seems to work for me @ricklove

**Comment by obstropolos** [comment]
A number of requests / feedback: 

- Please make sure to remove the `pnpm` dependencies and stick to the original package management using `yarn` 

Screen Shot 2021-04-04 at 8 18 36 PM

- Upon initially creating the transaction, we get the "in transit" message. Instead of this, would it be possible to have more context such as "Minting NFT?" or if we could even add the name to the message so the user knows what they're minting?

Screen Shot 2021-04-04 at 8 19 46 PM

- If a user rejects a transaction or a transaction fails, there's no context on these queue modals other than it being in transit forever. Do we have some kind of fallback if the transaction fails? 

Screen Shot 2021-04-04 at 8 20 52 PM

- After hitting the mint button in Temple and confirming the transaction, there's nothing that takes me away from the create asset page, leaving me to possibly hit the mint button a number of times again thinking that I haven't minted anything. Would we be able to redirect users back to their assets with the modal still there to let them know that it's complete? 

Screen Shot 2021-04-04 at 8 22 18 PM

- After my asset was created and I hit `New Asset`, it brings be back to my old asset cached. After the asset is created, I expect to be brought to the first step of the new asset screen. Additionally, there's a brand new flow created by @lambdahands to be pulled from the `develop` branch which might also affect this.

**Comment by maelswarm** [comment]
@obstropolos 

pnpm removed.

Redirect is now in place to  '/collections'

The other stuff

```Upon initially creating the transaction, we get the "in transit" message. Instead of this, would it be possible to have more context such as "Minting NFT?" or if we could even add the name to the message so the user knows what they're minting?```

This may have to go on a wish list?

**Comment by obstropolos** [comment]
Thanks for the changes. 

- It looks like the new asset is still caching (go to create a new asset while the current one is in transit) - is there a way to make the user restart the process to create another NFT if they wish to do so without having to hit 'cancel'? 
- Is there any way to at least change "In Transit?" because there's no context unfortunately for someone to point to when an asset is being created like this / knowing if they're actually creating an NFT. Even if we can change those two words. 
- Additionally, is there any way to make the queue modals a drop larger? They get a bit lost on the lower right side of the screen.

**Comment by lambdahands** [comment]
Closing in favor of #330

**Pull request closed without merging**